sql/core/src/main/scala/org/apache/spark/sql/execution/SparkStrategies.scala

```diff
@@ -282,14 +282,16 @@ abstract class SparkStrategies extends QueryPlanner[SparkPlan] {
 
       // Pick BroadcastNestedLoopJoin if one side could be broadcast
       case j @ logical.Join(left, right, joinType, condition, hint)
-          if canBroadcastByHints(joinType, left, right, hint) =>
-        val buildSide = broadcastSideByHints(joinType, left, right, hint)
+          if hint.leftHint.exists(_.broadcast) || hint.rightHint.exists(_.broadcast) =>
+        val buildLeft = hint.leftHint.exists(_.broadcast)
+        val buildRight = hint.rightHint.exists(_.broadcast)
```

Comment:
Our test cases do not cover the changes using these hints, right? Could you add them?